activerecord/lib/active_record/connection_adapters/connection_specification.rb

```diff
@@ -164,7 +164,7 @@ def resolve_all
         #   spec.config
         #   # => { "host" => "localhost", "database" => "foo", "adapter" => "sqlite3" }
         #
-        def spec(config)
+        def spec(config, id = nil)
```

Comment:
Find the use of `id` throughout confusing. Why not go with `specification_name`? That feels much closer to its nature of being symbols and strings like `:production`, `"primary"`. Put another way id reads too abstract to me, while name is more down to earth :smile: